src/main/scala/cromwell/binding/AstTools.scala

```diff
@@ -55,10 +60,21 @@ object AstTools {
           case "true" => WdlBoolean.True
           case "false" => WdlBoolean.False
         }
+        // TODO: The below cases, ArrayLiteral and MapLiteral are brittle.  They recursively call this wdlValue().
+        // However, those recursive calls might contain full-on expressions instead of just other literals.  This
+        // whole thing ought to be part of the regular expression evaluator, though I imagine that's non-trivial.
         case a: Ast if a.getName == "ArrayLiteral" && wdlType.isInstanceOf[WdlArrayType] =>
           val arrType = wdlType.asInstanceOf[WdlArrayType]
           val elements = a.getAttribute("values").asInstanceOf[AstList].asScala.toVector.map{node => node.wdlValue(arrType.memberType, wdlSyntaxErrorFormatter)}
           WdlArray(arrType, elements)
+        case a: Ast if a.getName == "MapLiteral" && wdlType.isInstanceOf[WdlMapType] =>
+          val mapType = wdlType.asInstanceOf[WdlMapType]
+          val elements = a.getAttribute("map").asInstanceOf[AstList].asScala.toVector.map { kvnode =>
```

Comment:
it might be nice to define a function which takes an AstNode and transforms it into the k->v and then just map over that